mods/ra/maps/fall-of-greece-2-evacuation/evacuation-AI.lua

```diff
@@ -0,0 +1,130 @@
+--[[
+   Copyright 2007-2021 The OpenRA Developers (see AUTHORS)
+   This file is part of OpenRA, which is free software. It is made
+   available to you under the terms of the GNU General Public License
+   as published by the Free Software Foundation, either version 3 of
+   the License, or (at your option) any later version. For more
+   information, see COPYING.
+]]
+AttackGroup = { }
+SovietInfantry = { "e1", "e2" }
+SovietVehicles = { "3tnk", "3tnk", "v2rl" }
+SovietAircraftType = { "yak" }
+Yaks = { }
+AttackPaths = { AttackRight, AttackLeft }
+
+AttackGroupSizes =
+{
+	easy = 8,
+	normal = 9,
+	hard = 10
+}
+
+ProductionInterval =
+{
+	easy = DateTime.Seconds(20),
+	normal = DateTime.Seconds(10),
+	hard = DateTime.Seconds(5)
+}
+
+SendAttackGroup = function()
+	if #AttackGroup < AttackGroupSize then
+		return
+	end
+
+	local path = Utils.Random(AttackPaths)
+	Utils.Do(AttackGroup, function(unit)
+		if not unit.IsDead then
+			unit.AttackMove(path.Location)
+			Trigger.OnIdle(unit, unit.Hunt)
+		end
+	end)
+
+	AttackGroup = { }
+end
+
+ProduceInfantry = function()
+	if USSRRax.IsDead or USSRRax.Owner ~= USSR then
+		return
+	end
+
+	USSR.Build({ Utils.Random(SovietInfantry) }, function(units)
+		table.insert(AttackGroup, units[1])
+		SendAttackGroup()
+		Trigger.AfterDelay(ProductionInterval[Map.LobbyOption("difficulty")], ProduceInfantry)
+	end)
+end
+
+ProduceVehicles = function()
+	if USSRWarFactory.IsDead or USSRWarFactory.Owner ~= USSR then
+		return
+	end
+
+	USSR.Build({ Utils.Random(SovietVehicles) }, function(units)
+		table.insert(AttackGroup, units[1])
+		SendAttackGroup()
+		Trigger.AfterDelay(ProductionInterval[Map.LobbyOption("difficulty")], ProduceVehicles)
+	end)
+end
+
+ProduceAircraft = function()
+	if Airfield.IsDead or Airfield.Owner ~= USSR then
+		return
+	end
+
+	USSR.Build(SovietAircraftType, function(units)
+		local yak = units[1]
+		Yaks[#Yaks + 1] = yak
+
+		Trigger.OnKilled(yak, ProduceAircraft)
+
+		local alive = Utils.Where(Yaks, function(y) return not y.IsDead end)
+		if #alive < 2 then
+			Trigger.AfterDelay(DateTime.Seconds(ProductionInterval[Map.LobbyOption("difficulty")] / 2), ProduceAircraft)
+		end
+
+		InitializeAttackAircraft(yak, Allies)
```

Comment:
I couldn't see where this is defined.

Reply:
`campaign-global.lua` in mods > ra > bits > scripts